apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts

```diff
@@ -4477,8 +4478,25 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
      */
     handleOutsideClick(): void {
         this.activeIndx = null;
+        this.checkVoucherEntries();
+    }
+
+    /** @memberOf ProformaInvoiceComponent
+     * check valid entry if have any then enable save button **/
+    checkVoucherEntries() {
+        if (this.invFormData.entries && this.invFormData.entries.length) {
+            let validLineItem;
+            for (let i = 0; i < this.invFormData.entries.length; i++) {
+                validLineItem = this.invFormData.entries[i].transactions.find(transaction => (transaction.accountUniqueName && transaction.accountUniqueName !== '' && transaction.amount > 0));
```

Comment:
Avoid redundant checks
```suggestion
                validLineItem = this.invFormData.entries[i].transactions.find(transaction => (transaction.accountUniqueName && transaction.amount > 0));
```